### PR TITLE
[GITHUB] build.yml: Use all matrix values to restore Linux build cache

### PR DESCRIPTION
**Pull request by SergeGautherie**
## Purpose

Sanity: do not mix different build configurations.

Addendum to f777e6bd and 732f223.

## Proposed changes

- Extend cache key.

### CODE REVIEW

**Comment by maj113** [comment]
@SergeGautherie seems that it is no longer finding a ccache hit with this PR:

```
Run actions/cache@v3
  with:
    path: ccache
    key: ccache-clang-amd64-Debug-0x502-b4da88ffca1173c09ea585579bc61a92fc269da9
    restore-keys: ccache-clang-amd64-Debug-0x502-
  
    enableCrossOsArchive: false
    fail-on-cache-miss: false
    lookup-only: false
  env:
    D_CLANG_VERSION: -DCLANG_VERSION=13
Cache not found for input keys: ccache-clang-amd64-Debug-0x502-b4da88ffca1173c09ea585579bc61a92fc269da9, ccache-clang-amd64-Debug-0x502-
```

which can be observed on all build-linux builds on this PR, master and other PRs don't have this issue

**Review by maj113 — APPROVED** [review]

**Comment by SergeGautherie** [comment]
> no longer

Well, old cache does not match new key.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -133,7 +133,7 @@ jobs:
       with:
         path: src
     - name: Configure
-      run: cmake -S src -B build -G Ninja -DCMAKE_TOOLCHAIN_FILE:FILEPATH=toolchain-msvc.cmake -DARCH:STRING=${{matrix.arch}} -DCMAKE_BUILD_TYPE=${{matrix.config}} -DENABLE_ROSTESTS=1 -DENABLE_ROSAPPS=1 -DDLL_EXPORT_VERSION=${{matrix.dllver}}
+      run: cmake -S src -B build -G Ninja -DCMAKE_TOOLCHAIN_FILE:FILEPATH=toolchain-msvc.cmake -DARCH:STRING=${{matrix.arch}} -DCMAKE_BUILD_TYPE=${{matrix.config}} -DDLL_EXPORT_VERSION=${{matrix.dllver}} -DENABLE_ROSTESTS=1 -DENABLE_ROSAPPS=1
```
Comment (JoachimHenze) [inline .github/workflows/build.yml:136]: What is the purpose/motivation of that parameters reordering?

Comment (SergeGautherie) [reply .github/workflows/build.yml:136]: Consistency with the other places, grouping all matrix values together...

**Comment by SergeGautherie** [comment]
@binarymaster, Would you review/commit this PR too?

**Review by binarymaster — APPROVED** [review]

**Review by HBelusca — APPROVED** [review]

**Pull request merged**